NIM : 42324046 

Nama : Ellyzhabeth

PRODI : DIII Teknologi Informasi

A. PRAKTIKUM

In [1]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_iris
# Load dataset Iris dari scikit-learn
iris = load_iris()

*Penjelasan :*
Kode ini melakukan import library yang akan digunakan selama proses klasifikasi. numpy atau np digunakan untuk mengolah data dalam bentuk array atau kumpulan angka. pandas atau pd digunakan untuk mengolah data dalam bentuk tabel atau DataFrame. Sedangkan load_iris digunakan untuk mengambil dataset Iris yang sudah disediakan oleh library Scikit-learn, sehingga tidak perlu membuat atau memasukkan dataset secara manual.

load_iris() digunakan untuk mengambil dataset Iris dari Scikit-learn. Dataset Iris merupakan dataset yang berisi data bunga Iris. Setiap data memiliki empat fitur, yaitu sepal length, sepal width, petal length, dan petal width. Dataset ini juga mempunyai tiga jenis atau spesies bunga, yaitu setosa, versicolor, dan virginica.

In [2]:
# Konversi data menjadi DataFrame Pandas
df = pd.DataFrame(iris.data, columns=iris.feature_names)
df["Species"] = iris.target_names[iris.target]
# Definisikan X (fitur) dan y (target/label)
X = df.drop(columns=["Species"])
y = df["Species"]
print(X)

     sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                  5.1               3.5                1.4               0.2
1                  4.9               3.0                1.4               0.2
2                  4.7               3.2                1.3               0.2
3                  4.6               3.1                1.5               0.2
4                  5.0               3.6                1.4               0.2
..                 ...               ...                ...               ...
145                6.7               3.0                5.2               2.3
146                6.3               2.5                5.0               1.9
147                6.5               3.0                5.2               2.0
148                6.2               3.4                5.4               2.3
149                5.9               3.0                5.1               1.8

[150 rows x 4 columns]


*Penjelasan:*
Data Iris diubah menjadi DataFrame Pandas agar lebih mudah diolah. Kemudian ditambahkan kolom Species yang berisi jenis bunga seperti setosa, versicolor, dan virginica. Setelah itu, data dibagi menjadi X dan y, di mana X berisi empat fitur bunga, sedangkan y berisi jenis/species bunga yang menjadi target prediksi. Terakhir, print(X) digunakan untuk menampilkan data fitur tersebut.

In [3]:
y

0         setosa
1         setosa
2         setosa
3         setosa
4         setosa
         ...    
145    virginica
146    virginica
147    virginica
148    virginica
149    virginica
Name: Species, Length: 150, dtype: str

*Penjelasa:*
Menunjukkan bahwa y berisi label atau nama spesies dari setiap data bunga Iris. Jadi, data nomor 0–4 termasuk setosa, sedangkan data nomor 145–149 termasuk virginica. Totalnya ada 150 data, dan Name: Species menunjukkan bahwa kolom tersebut bernama Species

In [4]:
from sklearn.preprocessing import LabelEncoder
lab = LabelEncoder()
y = lab.fit_transform(y)
y

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

*Penjelasan:*
Pada cell ini, mengimpor dan menggunakan LabelEncoder dari Scikit-Learn untuk melakukan konversi data kategori teks pada variabel y menjadi bentuk angka atau numerik. Proses fit_transform mengubah string nama spesies menjadi angka integer, contohnya setosa menjadi 0, versicolor menjadi 1, dan virginica menjadi 2. Mengubah label menjadi format angka merupakan langkah krusial agar model matematik dapat memproses data label tersebut.

In [5]:
from sklearn.model_selection import train_test_split
X_train, X_test = train_test_split(X, test_size=0.2, random_state=0)
y_train, y_test = train_test_split(y, test_size=0.2, random_state=0)

*Penjelasan:*
Fungsi train_test_split dari pustaka Scikit-Learn digunakan untuk membagi dataset menjadi data latih (training) dan data uji (testing). Pembagian ini dilakukan secara terpisah untuk variabel fitur X dan variabel target y. Parameter test_size=0.2 menentukan bahwa 20\% dari total data dialokasikan untuk pengujian (testing), sedangkan 80\% sisanya digunakan untuk pelatihan (training). Parameter random_state=0 dipasang agar proses pengacakan data menghasilkan urutan indeks yang sama setiap kali kode dijalankan, sehingga urutan baris data pada X_train tetap berpasangan secara presisi dengan y_train. 

In [6]:
from sklearn.naive_bayes import GaussianNB
model = GaussianNB()
NB_model = model.fit(X_train, y_train)

*Penjelasan:*
Kode ini, dilakukan impor algoritma GaussianNB dari Scikit-Learn. Algoritma Gaussian Naïve Bayes dipilih karena seluruh fitur dalam dataset Iris bersifat kontinu (angka riil/numerik) yang diasumsikan mengikuti distribusi normal (Gaussian). Selanjutnya, untuk melatih model tersebut menggunakan fungsi .fit() dengan memasukkan data latih X_train dan label latih y_train.

In [7]:
result = NB_model.predict(X_test)
result

array([2, 1, 0, 2, 0, 2, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 2, 1,
       0, 0, 2, 0, 0, 1, 1, 0])

*Penjelasan:*
Kode ini melakukan proses pengujian atau prediksi terhadap data baru yang belum pernah dilihat model sebelumnya, yaitu X_test. Fungsi .predict() akan mengembalikan array berisikan angka kelas (0, 1, atau 2) yang mewakili tebakan spesies model untuk setiap baris data dalam sampel uji X_test.

In [8]:
from sklearn.metrics import accuracy_score
print('Accuracy of data testing: ', accuracy_score(y_test, result)*100)

Accuracy of data testing:  96.66666666666667


*Penjelasan:*
Kode ini bertujuan untuk mengukur tingkat keberhasilan dan kinerja dari model klasifikasi yang telah dibuat. Dengan membandingkan nilai hasil prediksi (result) terhadap nilai target yang sebenarnya (y_test) menggunakan accuracy_score, diperoleh persentase akurasi model sebesar 96.67\% dalam memprediksi spesies secara tepat.

In [9]:
#prediksi inputan data baru
#input_data = pd.DataFrame(np.array([[5.1, 3.5, 1.4, 0.2]]), columns=iris.feature_names)
input_data = [[5.1, 3.5, 1.4, 0.2]]
result = NB_model.predict(input_data)
prediction_label = lab.inverse_transform(result)
print('Hasil prediksi untuk input baru adalah:', prediction_label)

Hasil prediksi untuk input baru adalah: ['setosa']


c:\Users\Ellyzhabeth Manurung\AppData\Local\Programs\Python\Python314\Lib\site-packages\sklearn\utils\validation.py:2827: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(


*Penjelasan:*
Kode ini mencoba melakukan prediksi terhadap satu sampel data baru menggunakan format array sederhana [[5.1, 3.5, 1.4, 0.2]]. Hasil prediksi berbentuk angka dikembalikan menjadi nama spesies asli menggunakan lab.inverse_transform() sehingga menghasilkan output setosa. Namun, cara ini memicu pesan peringatan (UserWarning) dari Scikit-Learn karena data dimasukkan dalam bentuk array tanpa memiliki nama kolom fitur.

In [10]:
input_data = pd.DataFrame(np.array([[5.1, 3.5, 1.4, 0.2]]), columns=iris.feature_names)
result = NB_model.predict(input_data)
prediction_label = lab.inverse_transform(result)
print('Hasil prediksi untuk input baru adalah:', prediction_label)

Hasil prediksi untuk input baru adalah: ['setosa']


*Penjelasan*
Kode ini merupakan perbaikan dari Cell 9 untuk menghilangkan pesan peringatan (warning). Data input baru dibungkus ke dalam DataFrame Pandas lengkap dengan nama kolom fitur yang sesuai dengan data saat pelatihan (columns=iris.feature_names). Dengan struktur data yang konsisten, model Gaussian Naïve Bayes dapat memprediksi input secara sempurna tanpa memunculkan peringatan, dan menghasilkan prediksi akhir yaitu setosa.

B.TUGAS

1. Ikuti semua tahapan dalam modul ini.

2. Prediksi data baru masuk ke species apa: [7.3, 3.8, 4.6, 2.3]

In [11]:
new_data = pd.DataFrame(
    np.array([[7.3, 3.8, 4.6, 2.3]]), columns=iris.feature_names
)

*Penjelasan:*
Kode ini untuk membuat variabel baru bernama new_data untuk menampung data inputan tugas. Nilai fitur [7.3, 3.8, 4.6, 2.3] diubah menjadi array NumPy dan dibungkus ke dalam DataFrame Pandas. Penggunaan DataFrame yang dilengkapi dengan nama kolom (columns=iris.feature_names) bertujuan agar struktur data baru ini identik dengan struktur dataset saat model dilatih, sehingga tidak memicu pesan peringatan (warning) dari Scikit-Learn.

In [12]:
result_new = NB_model.predict(new_data)

*Penjelasan:*
Kode ini menjalankan fungsi .predict() dari model NB_model (Gaussian Naïve Bayes) yang telah dilatih sebelumnya untuk mengklasifikasikan data pada new_data. Hasil dari proses ini adalah nilai prediksi berupa kode angka numerik (misalnya 2) yang mewakili indeks spesies tertentu.

In [13]:
prediction_label_new = lab.inverse_transform(result_new)

print('Hasil prediksi untuk data baru adalah:', prediction_label_new)

Hasil prediksi untuk data baru adalah: ['virginica']


*Penjelasan:*
fungsi lab.inverse_transform() digunakan untuk mengubah angka kode hasil prediksi pada result_new kembali menjadi nama label teks spesies aslinya. Setelah itu, perintah print() menampilkan teks hasil akhir ke layar, yang menunjukkan bahwa sampel data baru [7.3, 3.8, 4.6, 2.3] berhasil diprediksi tergolong ke dalam spesies virginica.